### ISSUE TITLE
remove "softmax" from DeepRank internal implementation

### ISSUE THREAD

**Issue by LilySnow** [DeepRank/deeprank#74]
`NeuralNet.py` does `softmax` internally. Shall we disable it? A user may want to add softmax to the last layer so that the loss function may benefit from it.

**Comment by LilySnow** [comment]
softmax is still hard coded. Why did you close the issue? @CunliangGeng
